keps/sig-cloud-provider/20190423-instance-host-labelling.md

```diff
@@ -0,0 +1,258 @@
+
+- [ ] kubernetes/enhancements issue in release milestone, which links to KEP (this should be a link to the KEP location in kubernetes/enhancements, not the initial KEP PR)
+- [ ] KEP approvers have set the KEP status to `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] Graduation criteria is in place
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+**Note:** Any PRs to move a KEP to `implementable` or significant changes once it is marked `implementable` should be approved by each of the KEP approvers. If any of those approvers is no longer appropriate than changes to that list should be approved by the remaining approvers and/or the owning SIG (or SIG-arch for cross cutting KEPs).
+
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://github.com/kubernetes/enhancements/issues
+[kubernetes/kubernetes]: https://github.com/kubernetes/kubernetes
+[kubernetes/website]: https://github.com/kubernetes/website
+
+## Summary
+
+Many kubernetes clusters are deployed on virtual machines (also referred to as
+"instances"), and information about the physical host for those instances (also
+referred to as "instance host") is not consistently available.  Instances which
+share a host can fail simultaneously if there is a problem with the host.  We
+propose to label nodes with their instance host ID so that controllers (namely,
+the scheduler) can decrease shared fate of pods from the same collection.
+
+## Motivation
+
+This KEP is motivated by the desire to improve reliability of workloads running
+in kubernetes clusters.  Today, the scheduler will attempt to spread pods from a
+given collection (e.g. a deployment) across nodes.  Even if every pod ends up on
+a different node, there may be multiple pods per instance host. This means that
+the deployment is not as reliable as it appears to be.
+
+By adding an instance host label, we can treat instance hosts as a failure
+domain, and perform spreading appropriately.
+
+
+### Goals
+
+- Introduce a well known label for instance host
+- Introduce a controller that keeps these labels updated
+
+### Non-Goals
+
+- Scheduling behavior that considers instance host
+- Controller behavior (e.g. deployment, statefulset) that considers instance
+  host
+- Any decision-making that is based on the instance host label
+
+## Proposal
```

Comment:
looks like the rest of this KEP is not written yet.

Reply:
it's better to only have context of this kep :)